master/buildbot/process/properties.py

```diff
@@ -720,6 +720,24 @@ def __init__(self, key, default=None, defaultWhenFalse=True):
         # Report on parent frame.
         _on_property_usage(key, stacklevel=1)
 
+    def __eq__(self, other):
+        return renderer(lambda props: props.getProperty(self.key) == other)
```

Comment:
I would render other, so that we can say Property("x") == Property("y")

```python
@implementer(IRenderable)
class _ComparaisonRenderer(util.ComparableMixin, object):

    compare_attrs = ('fn',)

    def __init__(self, v1, v2, cstr, comparator):
        self.v1, self.v2, self.comparator, self.cstr = v1, v2, comparator, cstr

    @defer.inlineCallbacks
    def getRenderingFor(self, props):
        v1 = yield props.render(self.v1)
        v2 = yield props.render(self.v2)
        defer.returnValue(self.comparator(v1, v2))

    def __repr__(self):
        return '%r %r %r' % (self.v1, self.cstr, self.v2)

def __eq__(self, other):
    return _ComparaisonRenderer(self, other, "==", lambda v1, v2: v1 == v2)
```